### PR TITLE
ocm: add new component repo (security-middleware)

### PR DESCRIPTION
**Pull request by dhaiducek**

### CODE REVIEW

**Review by stevekuznetsov — APPROVED** [review]

File: ci-operator/jobs/open-cluster-management/security-middleware/open-cluster-management-security-middleware-main-postsubmits.yaml
Comment (droslean) [inline ci-operator/jobs/open-cluster-management/security-middleware/open-cluster-management-security-middleware-main-postsubmits.yaml:1]: Why there is a postsubmit that runs the same test as the presubmit?

File: ci-operator/config/open-cluster-management/security-middleware/open-cluster-management-security-middleware-main.yaml
Comment (droslean) [inline ci-operator/config/open-cluster-management/security-middleware/open-cluster-management-security-middleware-main.yaml:21]: Why not having just one make command here instead?

File: ci-operator/jobs/open-cluster-management/security-middleware/open-cluster-management-security-middleware-main-postsubmits.yaml
Comment (ray-harris) [reply ci-operator/jobs/open-cluster-management/security-middleware/open-cluster-management-security-middleware-main-postsubmits.yaml:1]: This is not a test, it is an action. Sonar scans are run on PRs during the presubmit phase to check code quality before the PR is merged (and block the PR if it doesn't meet its quality gate. The scan results are published at sonarcloud.io.

After the PR is merged, the Sonar scan has to be run again, but this time on the branch that was just merged into. Again the results are published to sonarcloud.io.

File: ci-operator/config/open-cluster-management/security-middleware/open-cluster-management-security-middleware-main.yaml
Comment (ray-harris) [reply ci-operator/config/open-cluster-management/security-middleware/open-cluster-management-security-middleware-main.yaml:21]: We use a consolidated build harness for all 55+ of our repos instead of a different make file for each repo. Some repos need to make minor changes to the build harness settings or run multiple steps from the build harness.

**Pull request merged**
